lib/improver/blending/weights.py

```diff
@@ -493,13 +493,30 @@ def _create_new_weights_cube(self, cube, weights):
         for cube_slice, weight in (
                 zip(cube.slices_over(self.weighting_coord_name), weights)):
             sub_slice = cube_slice[..., 0, 0]
-            sub_slice.remove_coord(sub_slice.coord(axis='x'))
-            sub_slice.remove_coord(sub_slice.coord(axis='y'))
             sub_slice.data = np.ones(sub_slice.data.shape)*weight
             cubelist.append(sub_slice)
+
+        # re-order dimension coordinates to match input cube
         new_weights_cube = (
             check_cube_coordinates(cube[..., 0, 0], cubelist.merge_cube()))
+
+        # remove all scalar coordinates that are not time-, model- or
+        # blend-related
+        dim_coords = new_weights_cube.coords(dim_coords=True)
+        keep_coords = ["time", "forecast_period", "forecast_reference_time",
+                       "model_id", "model_configuration",
+                       self.weighting_coord_name, self.config_coord_name]
+        for coord in new_weights_cube.coords():
+            if coord not in dim_coords and coord.name() not in keep_coords:
```

Comment:
Do we want to capture any instances of a cube with a dimension coord of length 1? i.e one that would be demoted to a scalar using iris.utils.squeeze?

Reply:
Good question.  On investigation, I'm confident we want to keep all dimension coords.  The plugin slices the input cube over appropriate dimensions to create a template for the weights cube (see `_slice_input_cubes` and `_define_slice`, line ~550 of this module).  The only coordinates that should be present as dimensions on the initial weights cube are the weighting coord and two spatial coords.  We definitely want to retain all three of those on the weights cube.

Reply:
Thanks. This bit of code can be left alone then.